### PR TITLE
ConsoleActivity: Switch from ViewFlipper to ViewPager.

### PR DESCRIPTION
**Pull request by jklein24**
This simplifies the code for creating terminals, removes lots of code,
and totally eliminates the need for swipe handling.
From the UI side, the swipe now sticks to the user's finger and looks
much nicer. It will also make adding tabs much simpler.

### CODE REVIEW

**Comment by adb2001** [comment]
Nice work !

**Comment by kruton** [comment]
This doesn't update the input box in the correct window. I couldn't repro this, but it happened once:

1. Open local term
2. Go back to host list
3. Open SSH term
4. Page back and forth between the terminals
5. "Password:" dialog shows up on local term

Here's an easier one:

1. Open local term
2. Open SSH term
3. Wait for "Password:" dialog to show up
4. Go to menu and "Disconnect"
5. "Password:" dialog shows up on local term

**Comment by kruton** [comment]
Oh, it might be that the terminal is almost slid all the way over to the first window when the password prompt pops up.

File: app/src/main/java/org/connectbot/ConsoleActivity.java [hunk 1]
```diff
@@ -169,20 +166,13 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
-
-				// check to see if this bridge was requested
-				if (bridge == requestedBridge)
-					requestedIndex = currentIndex;
-			}
+			adapter.notifyDataSetChanged();
+			requestedIndex = bound.getBridges().indexOf(requestedBridge);
 
-			setDisplayedTerminal(requestedIndex);
+			pager.setCurrentItem(requestedIndex == -1 ? 0 : requestedIndex);
```
Comment (kruton) [inline app/src/main/java/org/connectbot/ConsoleActivity.java]: I think this is missing an `updatePromptVisible`, etc. Perhaps you should wrap all that in a new `setDisplayedTerminal` that calls `pager.setCurrentItem(…)` in addition to `updatePromptVisible()` and `updateEmptyVisible()`.

File: app/src/main/java/org/connectbot/ConsoleActivity.java [hunk 2]
```diff
@@ -985,15 +938,16 @@ protected void onNewIntent(Intent intent) {
...
 			} else {
-				final int flipIndex = getFlipIndex(requestedBridge);
+				final int flipIndex = bound.getBridges().indexOf(requestedBridge);
 				if (flipIndex > requestedIndex) {
 					requestedIndex = flipIndex;
 				}
 			}
 
-			setDisplayedTerminal(requestedIndex);
+			pager.setCurrentItem(requestedIndex);
```
Comment (kruton) [inline app/src/main/java/org/connectbot/ConsoleActivity.java]: I think this is also missing an `updatePromptVisible()`, etc., as above and should be wrapped in a call to `setDisplayedTerminal(…)`.

**Comment by jklein24** [comment]
OK I added the setDisplayedTerminal function back with the appropriate update calls, but am still able to repro this. The way I can repro it is if I open the ssh term and then swipe back to local before the prompt shows up. Looking into it more right now.

**Comment by jklein24** [comment]
OK, tracked it down. Turns out that the issue was in getCurrentTerminalView. It turns out the child index in the ViewPager != to the position according to the adapter. This means doing `pager.getChildAt(pager.getCurrentItem())` returns the wrong view sometimes. Tagging the views with their positions as they're inserted fixes the issue:

http://stackoverflow.com/questions/12854783/android-viewpager-get-the-current-view

**Comment by jklein24** [comment]
@adb2001, I'm working on adding tabs in https://github.com/jklein24/connectbot/tree/tabs, but probably only for large screens because they take up quite a bit of real estate.

**Comment by kruton** [comment]
No, this doesn't work either it seems. If you start a local terminal (gets tag `0`) then start an SSH session (gets tag `1`), wait for the prompt to come up, swipe back to the local session, select disconnect from the menu, the `getCurrentTerminalView()` will return `null` because the `pager.getCurrentItem()` returns `0` which ends up being null because the local term was tagged with `0`

**Comment by kruton** [comment]
Okay, I changed it to use `pager.getCurrentItem()` to query the `adapter` for what item is at that position. That works.

**Pull request merged**